### PR TITLE
[DCP] Automatically set `no_dist` if distributed is unavailable

### PR DESCRIPTION
**Pull request by LucasLLC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119335
* #119333
* #118894
* __->__#119332

Automatically set `no_dist` if distributed is unavailable

Differential Revision: [D53497641](https://our.internmc.facebook.com/intern/diff/D53497641/)

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

### CODE REVIEW

**Review by fegin — APPROVED** [review]
LGTM

File: torch/distributed/checkpoint/state_dict_loader.py [hunk 1]
```diff
@@ -131,6 +131,14 @@ def load(
         rank has an individual GPU, via ``torch.cuda.set_device()``.
     """
 
+    if no_dist is None:
+        no_dist = not (dist.is_available() and dist.is_initialized())
```
Comment (wconstab) [inline torch/distributed/checkpoint/state_dict_loader.py:135]: Helper fn for computing no_dist?

**Review by wconstab — APPROVED** [review]

**Review by wz337 — APPROVED** [review]
LGTM. 
I added the suppress-bc-linter. That should allow you pass the CI check.

**Comment by LucasLLC** [comment]
Fighting some CI issues, will circle back to this

**Comment by LucasLLC** [comment]
Closed in favor of https://github.com/pytorch/pytorch/pull/119816 due to CI issues

**Pull request closed without merging**